bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java

```diff
@@ -317,6 +319,14 @@ public String getLanguageDirection() {
             (request != null && StringUtils.isNotBlank(request.getParameter(GuideConstants.AF_DATA_REF)))) {
             formDataEnabled = true;
         }
+
+        // set draftId in properties in case of forms portal prefill
+        if (request != null && StringUtils.isNotBlank(request.getParameter(GuideConstants.AF_DATA_REF))) {
+            final String dataRef = request.getParameter(GuideConstants.AF_DATA_REF);
+            if (dataRef.startsWith(DRAFT_PREFILL_SERVICE)) {
```

Comment:
could dataRef be null?

Reply:
already have check for this in above if statement: StringUtils.isNotBlank(request.getParameter(GuideConstants.AF_DATA_REF)